### PR TITLE
Allow annotated literals inside as annotation args

### PR DESCRIPTION
**Pull request by gzm0**
This notably allows for silencing warnings inside annotation
arguments (see scala-js/scala-js#3737, ghik/silencer#32).

Fixes https://github.com/scala/bug/issues/11932

### CODE REVIEW

**Comment by som-snytt** [comment]
Is this https://github.com/scala/bug/issues/11932

**Review by dwijnand — APPROVED** [review]
LGTM

**Comment by gzm0** [comment]
> Is this scala/bug#11932

Yes.

**Comment by gzm0** [comment]
I have amended the commit with a reference to scala/bug#11932 and an additional test case with a type ascription.

What is the procedure for backporting things to 2.12.x? Cherry-pick and new PR? Target 2.12.x?

**Comment by sjrd** [comment]
> What is the procedure for backporting things to 2.12.x? Cherry-pick and new PR? Target 2.12.x?

If it is likely to merge cleanly, the best thing is to target 2.12.x. 2.12.x is periodically merged into 2.13.x.

**Comment by gzm0** [comment]
Rebased onto 2.12.x (no conflicts). ~~I do not understand what is happening with the GitHub merge conflict thingy...~~

File: src/reflect/scala/reflect/internal/AnnotationInfos.scala
Comment (sjrd) [inline src/reflect/scala/reflect/internal/AnnotationInfos.scala]:
```suggestion
    // We recurse over Typed / Annotated trees to allow things like:
```

**Review by sjrd — APPROVED** [review]

**Review by lrytz — APPROVED** [review]

**Comment by dwijnand** [comment]
@retronym yours if you want it in 2.12.13 or push it to 2.12.14.

Personally I'd defer it, otherwise it'll be in 2.12 but not 2.13 for a while...

**Comment by SethTisue** [comment]
> Personally I'd defer it, otherwise it'll be in 2.12 but not 2.13 for a while.

Good point, I didn't consider that when I milestoned it. Agree the misalignment would be confusing for users.

**Comment by gzm0** [comment]
What is the status on this? Is there anything needed from my end at this point?

**Comment by dwijnand** [comment]
It's waiting for 2.12.13 to be tagged so it can be merged to be in the version after (as well as be branch merged into 2.13.x for its version after.)

**Pull request merged**
